includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,512 @@
+<?php
+/**
+ * Fetches and formats data for AMP reader themes.
+ *
+ * @package AMP
+ * @since 1.6.0
+ */
+
+/**
+ * Class AMP_Reader_Themes.
+ *
+ * @since 1.6.0
+ */
+final class AMP_Reader_Themes {
+	/**
+	 * Formatted theme data.
+	 *
+	 * @var array
+	 */
+	private $themes;
+
+	/**
+	 * The name of the currently active theme.
+	 *
+	 * @var string
+	 */
+	private $current_theme_name;
+
+	/**
+	 * Whether themes can be installed in the current WordPress installation.
+	 *
+	 * @var bool
+	 */
+	private $can_install_themes;
+
+	/**
+	 * The default reader theme.
+	 *
+	 * @var string
+	 */
+	const DEFAULT_READER_THEME = 'classic';
+
+	/**
+	 * Status indicating a reader theme is active on the site.
+	 *
+	 * @var string
+	 */
+	const STATUS_ACTIVE = 'active';
+
+	/**
+	 * Status indicating a reader theme is installed but not active.
+	 *
+	 * @var string
+	 */
+	const STATUS_INSTALLED = 'installed';
+
+	/**
+	 * Status indicating a reader theme is not installed but is installable.
+	 *
+	 * @var string
+	 */
+	const STATUS_INSTALLABLE = 'installable';
+
+	/**
+	 * Status indicating a reader theme is not installed and can't be installed.
+	 *
+	 * @var string
+	 */
+	const STATUS_NON_INSTALLABLE = 'non-installable';
+
+	/**
+	 * Reader themes copied from JSON data freturn ed from Wordpress.org themes endpoint. See ::get_default_supported_reader_themes.
+	 * Note: Descriptions are shortened and made translatable in the AMP_Reader_Theme_REST_Controller.
+	 *
+	 * @var array
+	 */
+	const DEFAULT_READER_THEMES = [
```

Comment:
Rather than storing all of the data statically in the class here, what about fetching the information (other than the screenshots) from the WordPress.org themes API? This would then allow for the strings to be translated, which w don't have right now as it stands.

Reply:
Nevermind, I see you have anticipated this. We should move to do this in the future.